dxp/code_review.md

```diff
@@ -4,20 +4,86 @@ These are guidelines that the Frontend Infrastructure Team follows when developi
+3.  **Always provide a description.** In order to expedite review, give the reviewer(s) all the context they need to understand what your PR aims to do, and any analysis that you undertook to arrive at your solution (this can include a description of alternatives that you _didn't_ select, and why) (see [example](https://github.com/liferay-frontend/liferay-portal/pull/2)).
+4.  **Always provide a test plan.** This is part of the description, and consists of the steps you took to verify that your change is correct (see [example](https://github.com/liferay-frontend/liferay-portal/pull/34)). This serves two purposes: one is to enable the reviewer to easily check your PR (if they deem it necessary); the other is to provide the reviewer with information that may save them the effort of testing it themselves (that is, if your test plan is good, the reviewer can feel more confident about the change).
+5.  **Always assign a reviewer.** All team members are collaborators on the repo and may be assigned as reviewers. Proactively selecting reviewers and `@`-mentioning them will minimize the amount of time your pull sits unattended in the queue.
+6.  **Use the assignee to make current ownership clear.** If multiple people are involved in a pull, designate the person who is responsible for moving the pull to the next step as the assignee. For example, the assignee could be a person doing deep review of a change, or rebasing an outdated pull, or diagnosing CI failures; it need not be the author of the change.
+7.  **Use labels to make current status clear.** We have labels such as "awaiting response", "needs update", "forwarding", and so on to make it clear what the next steps are for each pull. Note that we even have "approved" and "changes requested" labels (which actually duplicate the indications that the GitHub UI provides) to make the status totally obvious.
+8.  **Ensure quality by analyzing test results.** At the time of writing, even a pull with a green CI result and "no unique failures" should be carefully inspected before running `ci:forward`, and specifically, "failures in common with upstream" should be examined to make sure that the pull does not introduce _additional_ failure that happens to overlap with existing failure in the upstream. If in doubt, ask QA for help, either in the `#d-quality-assurance` Slack channel, or directly mentioning our QA on GitHub or in the team Slack channel.
+9.  **Allow edits from maintainers.** When you create your pull, please be sure check the "Allow edits and access to secrets by maintainers" box at the bottom of the pull request form. This allows other collaborators on the repo to push to the pull's corresponding branch on your fork, thus updating the pull. This can save wasteful round trips, which can be expensive due to time-zone delays.
```

Comment:
Many people don't use GitHub UI to generate the PRs... I think [Inviting Collaborators](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/inviting-collaborators-to-a-personal-repository) might be easier... 🤔

Reply:
Everybody is already a collaborator:

![Screenshot 2020-06-19 -122714-5yjsDa6W@2x](https://user-images.githubusercontent.com/7074/85123258-38d2da80-b228-11ea-9af7-214343a495e8.png)

The only way that would help would be if people pushed their branches to *this* repo instead of their own. Is that what you're proposing?

Reply:
There are some risks to that (easy to push `master`, potentially submitting unwanted commits there).

Reply:
> The only way that would help would be if people pushed their branches to this repo instead of their own. Is that what you're proposing?

No, I was saying make everyone (or at least heavy reviewers) collaborators in THEIR own forks. That way, reviewers can push to personal forks keeping the `liferay-frontend` repo kinda safe...

Reply:
Sounds like a good idea, although it is a bit of duplication. Oh well... `#GitHubLife`

Reply:
Resolved by: https://github.com/liferay/liferay-frontend-guidelines/pull/167/commits/edf12b04ee13e36f599170f019a3d69b98928375